typing/parmatch.ml

```diff
@@ -886,30 +857,38 @@ let rec get_variant_constructors env ty =
     end
   | _ -> fatal_error "Parmatch.get_variant_constructors"
 
-(* Sends back a pattern that complements constructor tags all_tag *)
-let complete_constrs constr all_tags =
+module ConstructorSet = Set.Make(struct
+  type t = constructor_description
+  let compare c1 c2 = String.compare c1.cstr_name c2.cstr_name
+end)
+
+(* Sends back a pattern that complements the given constructors all_constrs *)
+let complete_constrs constr all_constrs =
   let c = constr.pat_desc in
-  let not_tags = complete_tags c.cstr_consts c.cstr_nonconsts all_tags in
   let constrs = get_variant_constructors constr.pat_env c.cstr_res in
+  let all_constrs = ConstructorSet.of_list all_constrs in
```

Comment:
The fact that `all_constrs` is a subset of `constrs` is confusing.

Reply:
Indeed, I was using the same naming as before... Changed to `used_constrs`.